src/wallet/wallet2.cpp

```diff
@@ -1752,8 +1782,13 @@ size_t wallet2::get_transfer_details(const crypto::key_image &ki) const
   for (size_t idx = 0; idx < m_transfers.size(); ++idx)
   {
     const transfer_details &td = m_transfers[idx];
-    if (td.m_key_image_known && td.m_key_image == ki)
-      return idx;
+    if (td.m_key_image == ki)
+    {
+      if (td.m_key_image_known)
+        return idx;
+      else if (td.m_key_image_partial)
+        CHECK_AND_ASSERT_THROW_MES(false, "Partial key image matches, but freezing partial key images is ambiguous");
+    }
```

Comment:
It theoretically shouldn't change the behavior, just provide a better error message on failure. Before this change, it would also fail if we matched a partial key image since we skip those transfers anyways but the error message would be "key image not found".

Reply:
The only time there would be a difference is if there was a wallet bug where a partial key image was marked as "known", or a full key image was marked as "partial". Inside the `process_new_transaction`, `m_key_image_known` is set as `!m_watch_only && !m_multisig`.

Reply:
I can't speak for the original author, but I think it's a UX thing. Since the key images provided to this function are largely provided directly from users, there's basically no scenario (that I can think of) which would warrant looking up partial key images. What's much more likely is someone sees a partial key image used somewhere (perhaps in the construction data), and thinks its a full key image and tries to use it as one and then gets unexpected behavior from the wallet.

Reply:
That error message is better though